src/vdom.ts

```diff
@@ -543,7 +578,7 @@ function updateChildren(
 	const oldChildrenLength = oldChildren.length;
 	const newChildrenLength = newChildren.length;
 	const transitions = projectionOptions.transitions!;
-
+	projectionOptions = { ...projectionOptions, ...{ depth: projectionOptions.depth + 1 } };
```

Comment:
No need for the second spread here or on line 671.
`projectionOptions = { ...projectionOptions, depth: projectionOptions.depth + 1 };`